### DISCUSSION TITLE
ENet NuGet package issues with UnrealCLR

### DISCUSSION THREAD

**Discussion by LeviLeckenby** [nxrighthere/UnrealCLR#71] [Help] [answered]
Hi, I've installed the Unreal CLR and am wanting to integrate the ENet-Charp nuget package as my networking solution. I've loaded it in via the Nuget Package manager, and the code is building to the (Project)/Manage folder. It works when I don't include ENet. But when I try and run it with ENet added, I get the error below. Does anyone know how I would go about getting this to work?

An assembly specified in the application dependencies manifest was not found:
package: 'ENet-CSharp', version: '2.4.5'
    path: 'runtimes/win/native/enet.dll'
   at System.Runtime.Loader.AssemblyDependencyResolver..ctor(String componentAssemblyPath)
   at UnrealEngine.Plugins.Loader.ManagedLoadContext..ctor(String mainAssemblyPath, IReadOnlyDictionary`2 managedAssemblies, IReadOnlyDictionary`2 nativeLibraries, IReadOnlyCollection`1 privateAssemblies, IReadOnlyCollection`1 defaultAssemblies, IReadOnlyCollection`1 additionalProbingPaths, IReadOnlyCollection`1 resourceProbingPaths, AssemblyLoadContext defa
ultLoadContext, Boolean preferDefaultLoadContext, Boolean isCollectible, Boolean loadInMemory, Boolean shadowCopyNativeLibraries)

**Comment by nxrighthere** [comment, marked as answer]
Last time I checked several NuGet packages, this one worked fine for me. Did you publish your project to the `%Project%/Managed` folder? If no, then assemblies are not copied into it. Make sure that libraries persist in the `%Project%/Managed` folder.

**Reply by LeviLeckenby** [reply]
You are right, my post-build event was a dotnet build into the manage dir, which only copied the enet.dll for some reason, and didn't copy the ENet-CSharp.dll. Doing a dotnet publish brings everything across. Thanks @nxrighthere ! That works great now 👍
